### DISCUSSION TITLE
Compressed GLB files crashing in Android and gltf_viewer.

### DISCUSSION THREAD

**Discussion by Mehequanna** [google/filament#6567] [Q&A]
Hello!

I've been playing around with compressed glbs and Filament on Android. When I try to use a compressed glb like the one attached, the Android app and Filament's gltf_viewer crash with the same errors:

Draco + KTX Compressed
```
FEngine (64 bits) created at 0x7f97c3872000 (threading is enabled)
FEngine resolved backend: Metal
Selected physical device 'Apple M1 Max'
FEngine feature level: 3
Missing texture provider for image\/ktx2
Missing texture provider for image\/jpeg
[1]    60697 segmentation fault  ./gltf_viewer 

```

Draco Compression
```
FEngine (64 bits) created at 0x7fce55855000 (threading is enabled)
FEngine resolved backend: Metal
Selected physical device 'Apple M1 Max'
FEngine feature level: 3
Missing texture provider for image\/jpeg
Missing texture provider for image\/jpeg
[1]    60725 segmentation fault  ./gltf_viewer 
```
[SHOE_CABINET_draco2.glb.zip](https://github.com/google/filament/files/10759459/SHOE_CABINET_draco2.glb.zip)

When I check the file with [Khronos GLTF Validator](https://github.khronos.org/glTF-Validator/), it says valid. So I'm wondering why Filament can't handle it. Does Filament support Draco and/or KTX compression?

Thanks for all the work you do!

**Comment by romainguy** [comment]
Draco and KTX are supported. Looks like a bug in our glTF loader.

**Comment by Mehequanna** [comment]
@romainguy Thanks for the reply and looking into this :)
